extensions/wikia/Search/WikiaSearchHelper.class.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+class WikiaSearchHelper {
+	const DEFAULT_LANG = 'en';
+	const CENTRAL_WIKI_SEARCH = '/wiki/Special:Search';
+
+	public function __construct() {
+		$this->wikiaLogoHelper = new WikiaLogoHelper();
+	}
+
+	/**
+	 * @desc This method appends /wiki/Special:Search to central URL.
+	 * It appends not localized version because SpecialPage::getTitle returns value based on content language
+	 * not user language.
+	 *
+	 * @param String $centralUrl - central wiki URL in given user language
+	 * @return string - url to Special:Search page
+	 */
+	public function getGlobalSearchUrl( $centralUrl ) {
+		return $centralUrl . self::CENTRAL_WIKI_SEARCH;
+	}
+
+	/**
+	 * @desc get language for search results.
+	 * If resultsLang param is set then use it if not get it from $wgLang
+	 *
+	 * @return String - language
+	 */
+	public function getLangForSearchResults() {
+		global $wgLanguageCode, $wgRequest;
+
+		$resultsLang = $wgRequest->getVal( 'resultsLang' );
+
+		return $resultsLang ? $resultsLang : $wgLanguageCode;
```

Comment:
Can be shortened to `return $resultsLang ?: $wgLanguageCode;`. This could possibly even just be `return $wgRequest->getVal( 'resultsLang', $resultsLang );`